.github/workflows/ci-base.yaml

```
@@ -111,10 +145,29 @@ jobs:
           version: '~> v2'
           args: ${{ env.goreleaser_args }}
           workdir: distributions/${{ inputs.distribution }}
+      
+      - name: Skip GoReleaser build (cached)
+        if: steps.cache-goreleaser.outputs.cache-hit == 'true'
+        run: echo "✅ GoReleaser build skipped - using cached binaries"
 
-      - name: Extract relevant metadata
+      - name: Extract relevant metadata from GoReleaser output
         run: |
-          VERSION=$(echo '${{ steps.goreleaser.outputs.metadata }}' | jq -r '.version')
+          if [ "${{ steps.cache-goreleaser.outputs.cache-hit }}" = "true" ]; then            
+            # Try to find version from dist folder structure or metadata file
+            if [ -f "distributions/${{ inputs.distribution }}/dist/metadata.json" ]; then
+              VERSION=$(jq -r '.version' distributions/${{ inputs.distribution }}/dist/metadata.json)
+            else
+              # Fallback version
+              VERSION="cached-$(date +%Y%m%d-%H%M)"
+              echo "⚠️ Warning: No metadata found, using fallback version: $VERSION"
```

Have we encountered this? Is this to have a fallback when goreleaser changes where it writes its metadata or is there a different scenario I am not thinking of? I am wondering if we should just fail the build instead of semi-silently continue?